lib/manager/sbt/readme.md

```diff
@@ -0,0 +1,245 @@
+E.G. https://astexplorer.net/#/gist/027dce11e6927b4ad39ea097ce6289b9/ea4d1048f32063a71d727d01178ae6d01087a62f
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+Dependencies can have a "scope" like `Test` or `Provided`.
+
+Renovate shouldn't care of scopes: they should be kept as is but they don't have any impact on the "new dependency resolution".
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+##### build.sbt
+
+Most common syntaxes:
+
+```scala
+// For Scala dependency (notice the double %%)
+libraryDependencies += "com.typesafe.play" %% "play-ws" % "2.6.10"
+// For Java dependency
+libraryDependencies += "org.mockito" % "mockito-all" % "1.10.18"
+
+// With a scope
+libraryDependencies += "org.mockito" % "mockito-all" % "1.10.18" % Test
+
+// Multiple dependencies at the same time
+libraryDependencies ++= Seq(
+  "com.typesafe.play" %% "play-ws" % "2.6.10",
+  "org.mockito" % "mockito-all" % "1.10.18"
+)
+
+// Defining some dependency overriding (same as for build.sbt)
+dependencyOverrides += "com.google.guava" % "guava" % "23.0"
+// Multiple at once
+dependencyOverrides ++= Seq(
+  "org.scala-lang.modules" %% "scala-parser-combinators" % "1.1.0",
+  "com.google.guava" % "guava" % "23.0"
+)
+```
+
+There is two variables to look for `libraryDependencies` and `dependencyOverrides`.
+
+More custom syntaxes with plain Scala code:
+
+```scala
+val myDependencyVersion = "2.6.10"
+val myDependencyGroupId = "com.typesafe.play"
+val myDependencyArtifactId = "play-json"
+val myDependency: ModuleID = myDependencyGroupId %% myDependencyArtifactId % myDependencyVersion
+
+libraryDependencies += myDependency
+```
+
+You could imagine some `if` statements as well as any other Scala syntax but I think it's pretty rare.
+
+##### plugins.sbt
+
+Most common syntaxes:
+
+```scala
+// Declaring a plugin
+addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.9.0")
+
+// Defining some dependency overriding (same as for build.sbt)
+dependencyOverrides += "org.webjars" % "webjars-locator-core" % "0.33"
+// Multiple at once
+dependencyOverrides ++= Seq(
+  "org.webjars" % "webjars-locator-core" % "0.33",
+  "org.codehaus.plexus" % "plexus-utils" % "3.0.17",
+  "com.google.guava" % "guava" % "23.0"
+)
+```
+
+---
+
+#### Describe which types of dependencies above are supported and which will be implemented in future:
+
+All to be supported.
+
+## Versioning
+
+#### What versioning scheme do the package files use?
+
+Ivy (https://ant.apache.org/ivy/history/2.3.0/ivyfile/dependency.html#revision).
+
+---
+
+#### Does this versioning scheme support range constraints, e.g. `^1.0.0` or `1.x`?
+
+Yes. With syntax like `[1.0,2.0]`. See https://ant.apache.org/ivy/history/2.3.0/ivyfile/dependency.html#revision.
+
+Maven version ranges are supported as well (https://www.scala-sbt.org/1.x/docs/Library-Dependencies.html#Ivy+revisions).
```

Comment:
How can we tell whether it's an ant/ivy syntax or maven syntax? And can you give an example in the earlier section of a full declaration using both?

Reply:
Actually, only the Ivy syntax is supported (I added examples). The link just explains how a Maven dependency having itself range versions defined for some dependency will be handled by sbt.